install_cluster/tasks/main.yml

```diff
@@ -3,27 +3,32 @@
 
 # TODO: Validate cluster settings
 
+- name: install cluster from cluster.yml
+  include_tasks: cluster.yml
+  when: sno | bool == True
+
 - name : Count the hosts to be discovered
   set_fact:
-    inventory_hosts: "{{ groups['masters'] | length + groups['workers'] | length }}"
+    inventory_hosts: "{{ groups['masters'] }}"
+  when: sno | bool == False
 
 - name: Join list for workers and masters
   set_fact:
-    nodes: "{{ groups['masters'] + groups['workers'] | default([]) }}"
-    inventory_nodes: "{{ groups['masters'] + groups['workers'] | default([]) }}"
+    nodes: "{{ groups['masters'] | default([]) }}"
+    inventory_nodes: "{{ groups['masters'] | default([]) }}"
+  when: sno | bool == False
 
-# Monitor hosts discovery
 - name: Wait for 20 min until all the hosts are discovered
-  uri:
-    url: "{{ URL_ASSISTED_INSTALLER_CLUSTER }}"
-    method: GET
-    status_code: [200, 201]
-    return_content: True
+  sonofspike.ocm.discover_host:
+    access_token: "{{ access_token }}"
+    api_token: "{{ API_TOKEN }}"
+    url_assisted_installer_cluster: "{{ URL_ASSISTED_INSTALLER_CLUSTER }}"
   register: cluster
   until: ( cluster.json.hosts | length == inventory_nodes | length ) and ( cluster.json.status == "pending-for-input")
   retries: 20
   delay: 60
-  when: install | bool == True
+  when: install | bool == True and sno | bool == False
```

Comment:
There seems to be alot of these is it worth having an `install_sno_cluster` role or perhaps two entirely different code paths?